provider/cluster/kube/builder.go

```diff
@@ -944,11 +944,56 @@ func ingressHost(lid mtypes.LeaseID, svc *manifest.Service) string {
 	return strings.ToLower(base32.HexEncoding.WithPadding(base32.NoPadding).EncodeToString(uid))
 }
 
+func (b *ingressBuilder) annotations() map[string]string {
+	// For kubernetes/ingress-nginx
+	// https://github.com/kubernetes/ingress-nginx
+	const root = "nginx.ingress.kubernetes.io"
+
+	httpOptions := b.expose.HTTPOptions
+
+	result := map[string]string{
+
+		fmt.Sprintf("%s/proxy-send-timeout", root):        fmt.Sprintf("%dms", httpOptions.SendTimeout),
+		fmt.Sprintf("%s/proxy-read-timeout", root):        fmt.Sprintf("%dms", httpOptions.ReadTimeout),
```

Comment:
should we just have them input it as a string `"100ms"`?

I wonder what will happen if we have invalid values here.

Reply:
The input type is `uint32` so they can't really input anything invalid. If they input zero we use the default. A positive number is a valid value, up to the limit.